chef/cookbooks/keystone/providers/register.rb

```
@@ -107,6 +107,38 @@
   end
 end
 
+# :add_domain_role specific attributes
+# attribute :domain_name, :kind_of => String
+# attribute :user_name, :kind_of => String
+# attribute :role_name, :kind_of => String
+action :add_domain_role do
+  http, headers = _build_connection(new_resource)
+
+  # get user_id
+  path = "/v2.0/users"
+  dir = "users"
+  user_id, uerror = _find_id(http, headers, new_resource.user_name, path, dir)
+  # get role_id
+  path = "/v2.0/OS-KSADM/roles"
+  dir = "roles"
+  role_id, rerror = _find_id(http, headers, new_resource.role_name, path, dir)
+  # get domain_id
+  path = "/v3/domains"
+  dir = "domains"
+  domain_id, derror = _find_id(http, headers, new_resource.domain_name, path, dir)
+
+  if uerror || rerror || derror
```

_find_id already logs a full message for any errors found, so unless this is for performance reasons (fail fast?) I dont think there is too much value into doing the check just after the call.